tests/test_tag_grants_with_mesh.py

```diff
@@ -0,0 +1,77 @@
+import tempfile
+import shutil
+import pickle
+import csv
+import os
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.preprocessing import MultiLabelBinarizer
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.linear_model import SGDClassifier
+
+from grants_tagger.tag_grants_with_mesh import tag_grants_with_mesh
+
+X = [
+    "all",
+    "one two",
+    "two",
+    "two hundred",
+    "one thousand"
+]
+Y = [
+    [str(i) for i in range(5000)],
+    ["1", "2"],
+    ["2"],
+    ["200"],
+    ["1000"]
+]
+
+def train_test_model(model_path, label_binarizer_path):
+    tfidf = TfidfVectorizer()
+    tfidf.fit(X)
+    with open(f"{model_path}/tfidf.pkl", "wb") as f:
+        f.write(pickle.dumps(tfidf))
+
+    X_vec = tfidf.transform(X)
+
+    label_binarizer = MultiLabelBinarizer()
+    label_binarizer.fit(Y)
+    with open(f"{label_binarizer_path}", "wb") as f:
+        f.write(pickle.dumps(label_binarizer))
+        f.seek(0)
+
+    Y_vec = label_binarizer.transform(Y)
+
+    for tag_i in range(0, Y_vec.shape[1], 512):
+        model = OneVsRestClassifier(SGDClassifier(loss='log'))
+        model.fit(X_vec, Y_vec[:, tag_i:tag_i+512])
+        with open(f"{model_path}/{tag_i}.pkl", "wb") as f:
+            f.write(pickle.dumps(model))
+
+def test_tag_grants_with_mesh():
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        model_path = f"{tmp_dir}/model/"
+        os.mkdir(model_path)
+        label_binarizer_path = f"{tmp_dir}/label_binarizer.pkl"
+        train_test_model(model_path, label_binarizer_path)
+
+        with tempfile.NamedTemporaryFile(mode="w+") as tmp_grants:
```

Comment:
You could potentially ditch using NamedTemporaryFile here as you've alread created a TemporaryDirectory above , in order to simplify things. Just create your file in the already created Temporary Directory and it will get cleaned up when the encapsulating context passes out of scope (i.e. the tmp_dir variable).

Admittedly it would only save a little overhead but you would then not have to reason about temp files as anything in that tmp_dir woudl get wiped out anyway.

Reply:
Good point, this is what i did in the other test and i was wondering whether it was hacky or standard.

Reply:
You can do it the way you've currently done (i.e. leave as is) if you want, it's just you can trim the excess overhead of managing another temp handle and reduce the amount of cleanup required by the library at the end of it's run by using the dir. I don't think there's a whole lot of guidance on whether it's better to do it the way you have or to use re-use the dir. Just wanted to point out that you could have.

Reply:
c1e0c692c941b7dc826aeaa287f99a59d9d3ca85